examples/demo-stf/src/tests/da_simulation.rs

```diff
@@ -14,43 +13,44 @@ use crate::runtime::Runtime;
 
 type C = DefaultContext;
 
-pub fn simulate_da(
-    value_setter_admin: DefaultPrivateKey,
-    election_admin: DefaultPrivateKey,
-) -> Vec<RawTx> {
+pub fn simulate_da(value_setter_admin: DefaultPrivateKey) -> Vec<RawTx> {
     let mut messages = Vec::default();
 
-    let election: ElectionCallMessages<C> = ElectionCallMessages::new(election_admin);
-    messages.extend(election.create_raw_txs::<Runtime<C>>());
+    let bank_generator = BankMessageGenerator::<C>::default();
+    let bank_txns = bank_generator.create_raw_txs::<Runtime<C>>();
 
     let value_setter = ValueSetterMessages::new(vec![ValueSetterMessage {
         admin: Rc::new(value_setter_admin),
         messages: vec![99, 33],
     }]);
     messages.extend(value_setter.create_raw_txs::<Runtime<C>>());
-
+    messages.extend(bank_txns);
     messages
 }
 
-pub fn simulate_da_with_revert_msg(election_admin: DefaultPrivateKey) -> Vec<RawTx> {
-    let election = InvalidElectionCallMessages::new(election_admin);
-    election.create_raw_txs::<Runtime<C>>()
-}
-
-pub fn simulate_da_with_bad_sig(election_admin: DefaultPrivateKey) -> Vec<RawTx> {
-    let election = BadSigElectionCallMessages::<DefaultContext>::new(election_admin);
-    election.create_raw_txs::<Runtime<C>>()
+pub fn simulate_da_with_revert_msg() -> Vec<RawTx> {
+    let mut messages = Vec::default();
+    let bank_generator = BankMessageGenerator::<C>::create_invalid_transfer();
+    let bank_txns = bank_generator.create_raw_txs::<Runtime<C>>();
+    messages.extend(bank_txns);
+    messages
 }
 
-// TODO: Remove once we fix test with bad nonce
-//   https://github.com/Sovereign-Labs/sovereign-sdk/issues/235
-#[allow(unused)]
-pub fn simulate_da_with_bad_nonce(election_admin: DefaultPrivateKey) -> Vec<RawTx> {
-    let election = BadNonceElectionCallMessages::new(election_admin);
-    election.create_raw_txs::<Runtime<C>>()
+pub fn simulate_da_with_bad_sig() -> Vec<RawTx> {
+    let b = BadSignatureBankCallMessages::new();
+    b.create_raw_txs::<Runtime<C>>()
 }
-
-pub fn simulate_da_with_bad_serialization(election_admin: DefaultPrivateKey) -> Vec<RawTx> {
-    let election = BadSerializationElectionCallMessages::new(election_admin);
-    election.create_raw_txs::<Runtime<C>>()
+//
+// // TODO: Remove once we fix test with bad nonce
+// //   https://github.com/Sovereign-Labs/sovereign-sdk/issues/235
+// #[allow(unused)]
+// pub fn simulate_da_with_bad_nonce(election_admin: DefaultPrivateKey) -> Vec<RawTx> {
```

Comment:
Is there anything right now stops us from testing this?

Reply:
ah good catch, i missed that one. the function isn't used, but right now I think we can add a test for that since we have a bad nonce error message and txn failure!